ZFSin/zfs/module/zfs/zfs_vnops_windows.c

```diff
@@ -342,6 +342,86 @@ int zfs_find_dvp_vp(zfsvfs_t *zfsvfs, char *filename, int finalpartmaynotexist,
 
 
 
+/*
+ * Call vnode_setunlink if zfs_zaccess_delete() allows it
+ * TODO: provide credentials
+ */
+NTSTATUS zfs_setunlink(vnode_t *vp, vnode_t *dvp) {
+
+	NTSTATUS Status = STATUS_UNSUCCESSFUL;
+
+	if (vp == NULL) {
+		Status = STATUS_INVALID_PARAMETER;
+		goto err;
+	}
+
+	znode_t *zp = NULL;
+	znode_t *dzp = NULL;
+	zfsvfs_t *zfsvfs;
+	VN_HOLD(vp);
+	zp = VTOZ(vp);
+
+	if (vp && zp) {
+		zfsvfs = zp->z_zfsvfs;
+	} else {
+		Status = STATUS_INVALID_PARAMETER;
+		goto err;
+	}
+
+	// if dvp == null, find it
+
+	if (dvp == NULL) {
+		uint64_t parent;
+
+		if (sa_lookup(zp->z_sa_hdl, SA_ZPL_PARENT(zfsvfs),
+			&parent, sizeof(parent)) != 0) {
+			goto err;
+		}
+		if (zfs_zget(zfsvfs, parent, &dzp)) {
+			dvp = NULL;
+			goto err;
+		}
+		dvp = ZTOV(dzp);
+	} else {
+		dzp = VTOZ(dvp);
+		VN_HOLD(dvp);
+	}
+
+	if (!dvp) {
```

Comment:
Probably not needed, since we deal with == NULL case above...

but, you dont need to make a new PR each time, you can make changes, `git commit --amend` it then `git push -f` to force push over the commits in the PR.

You dont need to fix this, unless you want to practice :)

Reply:
Now that you mention it, this is maybe a bit over-cautious....but some git practice sounds always good, give me a sec ;)

Reply:
ah, seems like it worked :) :+1: